### ISSUE TITLE
.search method ignores rails scoping

### ISSUE THREAD

**Issue by JFimex** [sunspot/sunspot#695]
(using sunspot_rails 2.1.1, and sunspot_solr 2.1.1)
Suppose you have these models 

class User < ActiveRecord::Base
    has_many :posts
    has_many :comments, through :posts
end

class Post < ActiveRecord::Base
    has_many :comments
end

class Comment
    searchable do
        text :content
    end
end

Create a few dummy comments and found this:

usr = User.first
usr.comments.count  #=> 0
usr.comments.search.results.count #=> 2
In other words if you search from within a subset of comments, you still searching from all comments.

**Comment by JFimex** [comment]
closing: no point raising an issue that can be fixed with 6 lines of code:


    class ActiveRecord::Base
	    def self.scoped_solr_search(&block)
		    ids = self.solr_search_ids(&block)
		    where(id: ids)
	    end
    end
